frontend/src/DiagramEditor/DiagramEditor.jsx

```diff
@@ -178,20 +234,97 @@ export class DiagramEditor extends React.Component {
     });
   };
 
-  onComponentMouseDown = (e, selectedComponentId) => {
+  onConnectionMouseDown = (e, connectionId) => {
+    this.setState({
+      selectedConnectionId: connectionId,
+    });
+  };
+
+  onComponentMouseDown = (e, componentId) => {
+    const { isConnecting } = this.state;
+    if (!isConnecting) {
```

Comment:
I guess if it is connecting, then we don't want to get down to the dragging logic below, so maybe we should have a quick escape on `if (isConnecting) {return}`

Reply:
I've done just that